### PR TITLE
Desktop: Log In & Renaming src into projects

### PR DESCRIPTION
**Pull request by jvnm-dev**
# PR Details

Integrating log in into the desktop app
Renaming /src into /projects

## Description

I'm integrating the log in interface using material UI & final-form.
I renamed /src folder into /projects because it was confusing

## Related Issue

#2 

## How Has This Been Tested

Coming soon

## Types of changes

- ~~Docs change / refactoring / dependency upgrade~~
- ~~Bug fix (non-breaking change which fixes an issue)~~
- [x] New feature (non-breaking change which adds functionality)
- ~~Breaking change (fix or feature that would cause existing functionality to change)~~

## Checklist

- [x] My code follows the code style of this project.
- ~~My change requires a change to the documentation.~~
- ~~I have updated the documentation accordingly.~~
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] My code is commented
- [x] All new and existing tests passed.

### CODE REVIEW

File: projects/desktop/src/assets/locales/EN.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (Sylv11) [inline projects/desktop/src/assets/locales/EN.json:1]: Add trailing coma in tslint.json ?

File: projects/desktop/src/components/LoginScreen/form/CheckboxField.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    const [t] = useTranslation()
+
+    return (
+        <>
+            <Field
+                name='remember'
+                component='input'
+                type='checkbox'
+            >
+                {({input}) => (
```
Comment (Sylv11) [inline projects/desktop/src/components/LoginScreen/form/CheckboxField.tsx:16]: `input` from props ? No interface ?

File: projects/desktop/src/components/LoginScreen/form/EmailField.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+export const EmailField = (): JSX.Element => {
+    const [t] = useTranslation()
+
+    return (
+        <Field
+            name='email'
+            type='email'
+            validate={composeValidators(required, validateEmail)}
+        >
+            {({ input, meta }) => (
```
Comment (Sylv11) [inline projects/desktop/src/components/LoginScreen/form/EmailField.tsx:16]: `input` and `meta` from props ? No interface ?

File: projects/desktop/src/components/LoginScreen/form/CheckboxField.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+import {Field} from 'react-final-form'
+import {useTranslation} from 'react-i18next'
+import {CheckboxContainer, CheckboxInput, CheckboxCheckmark} from '../styles'
+
+export const CheckboxField = () => {
+    const [t] = useTranslation()
```
Comment (Sylv11) [inline projects/desktop/src/components/LoginScreen/form/CheckboxField.tsx]: No type for `t` ?

File: projects/desktop/src/components/LoginScreen/form/EmailField.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+import { Field } from 'react-final-form'
+import { Input } from './'
+import { composeValidators, required, validateEmail } from '../../../constants/validators'
+import { useTranslation } from 'react-i18next'
+
+export const EmailField = (): JSX.Element => {
+    const [t] = useTranslation()
```
Comment (Sylv11) [inline projects/desktop/src/components/LoginScreen/form/EmailField.tsx]: No type for `t` ?

File: projects/desktop/src/components/LoginScreen/form/Form.tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+import { EmailField, PasswordField, CheckboxField } from './'
+import {CenteredForm, FormLink, GradientButton} from '../styles'
+import { useTranslation } from 'react-i18next'
+
+export const Form = ({handleSubmit}: {handleSubmit: () => void}): JSX.Element => {
+    const [t] = useTranslation()
```
Comment (Sylv11) [inline projects/desktop/src/components/LoginScreen/form/Form.tsx]: No type for `t` ?

File: projects/desktop/src/components/LoginScreen/form/Form.tsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+import { EmailField, PasswordField, CheckboxField } from './'
+import {CenteredForm, FormLink, GradientButton} from '../styles'
+import { useTranslation } from 'react-i18next'
+
+export const Form = ({handleSubmit}: {handleSubmit: () => void}): JSX.Element => {
```
Comment (Sylv11) [inline projects/desktop/src/components/LoginScreen/form/Form.tsx]: Interface for props

File: projects/desktop/src/components/LoginScreen/form/Input.tsx [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+} from '@material-ui/core'
+import {
+    VisibilityOff,
+    Visibility,
+} from '@material-ui/icons'
+import { colors } from '../../../constants'
+import { InputError } from '../styles'
+import { useTranslation } from 'react-i18next'
+
+const initialState = {
```
Comment (Sylv11) [inline projects/desktop/src/components/LoginScreen/form/Input.tsx]: Interface for initialState

File: projects/desktop/src/components/LoginScreen/form/Input.tsx [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+    visible: false,
+}
+
+const actions = {
```
Comment (Sylv11) [inline projects/desktop/src/components/LoginScreen/form/Input.tsx]: Type

File: projects/desktop/src/components/LoginScreen/form/Input.tsx [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+    TOGGLE_VISIBILITY: 'TOGGLE_VISIBILITY',
+}
+
+const reducer = (state: IStringAnyMap, action: IAction) => {
```
Comment (Sylv11) [inline projects/desktop/src/components/LoginScreen/form/Input.tsx]: Return function type

File: projects/desktop/src/components/LoginScreen/form/PasswordField.tsx [hunk 11]
```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+import { Field } from 'react-final-form'
+import { Input } from './'
+import { composeValidators, required, validateMinimumLength } from '../../../constants/validators';
+import { useTranslation } from 'react-i18next'
+
+export const PasswordField = (): JSX.Element => {
+    const [t] = useTranslation()
```
Comment (Sylv11) [inline projects/desktop/src/components/LoginScreen/form/PasswordField.tsx]: Type for `t`

File: projects/desktop/src/components/LoginScreen/form/PasswordField.tsx [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+    
+    return (
+        <Field
+            name='password'
+            type='password'
+            validate={composeValidators(required, validateMinimumLength(8))}
+        >
+            {({ input, meta }) => (
```
Comment (Sylv11) [inline projects/desktop/src/components/LoginScreen/form/PasswordField.tsx:16]: Interface for props ?

File: projects/desktop/src/components/LoginScreen/index.tsx [hunk 13]
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import {BigTitle, LoginLanguageChooser, Logo} from './styles'
+import { Container, LeftSide, RightSide } from './parts'
+import { Form as FinalForm } from 'react-final-form'
+import { Form } from './form'
+import { useTranslation } from 'react-i18next'
+import logo from '../../assets/img/logo-small.png'
+
+export const LoginScreen = (): JSX.Element => {
+    const [t] = useTranslation()
```
Comment (Sylv11) [inline projects/desktop/src/components/LoginScreen/index.tsx]: Type for `t` ?

File: projects/desktop/src/components/LoginScreen/index.tsx [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+
+    const getTitleContent = (): {__html: string} => {
```
Comment (Sylv11) [inline projects/desktop/src/components/LoginScreen/index.tsx]: Return function type

File: projects/desktop/src/components/LoginScreen/index.tsx [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+
+    return (
+        <Container>
+            <LeftSide>
+                <Logo src={logo} />
+                <LoginLanguageChooser />
+                <BigTitle dangerouslySetInnerHTML={getTitleContent()} />
+
+                <FinalForm
+                    onSubmit={values => console.log(values)}
```
Comment (Sylv11) [inline projects/desktop/src/components/LoginScreen/index.tsx:24]: the `console.log(values)` has to stay here ?

File: projects/desktop/src/components/common/LanguageChooser/LanguageChooser.tsx [hunk 16]
```diff
@@ -0,0 +1,58 @@
+import React, {useReducer} from 'react'
+import { Button } from '@material-ui/core'
+import {LanguageChooserDialog} from './LanguageChooserDialog'
+import { useTranslation } from 'react-i18next'
+
+const initialState = {
```
Comment (Sylv11) [inline projects/desktop/src/components/common/LanguageChooser/LanguageChooser.tsx]: Interface for initialState

File: projects/desktop/src/components/common/LanguageChooser/LanguageChooser.tsx [hunk 17]
```diff
@@ -0,0 +1,58 @@
...
+    open: false,
+}
+
+const actions = {
```
Comment (Sylv11) [inline projects/desktop/src/components/common/LanguageChooser/LanguageChooser.tsx]: Add types

File: projects/desktop/src/components/common/LanguageChooser/LanguageChooser.tsx [hunk 18]
```diff
@@ -0,0 +1,58 @@
...
+    TOGGLE_DIALOG: 'TOGGLE_DIALOG',
+}
+
+const reducer = (state: typeof initialState, action: IAction) => {
```
Comment (Sylv11) [inline projects/desktop/src/components/common/LanguageChooser/LanguageChooser.tsx]: Return function type ?

File: projects/desktop/src/components/common/LanguageChooser/LanguageChooser.tsx [hunk 19]
```diff
@@ -0,0 +1,58 @@
...
+            return {
+                ...state,
+                open: !state.open,
+            }
+        default:
+            throw new Error('Action type not found')
+    }
+}
+
+export const LanguageChooser = ({className}: {className?: string}) => {
```
Comment (Sylv11) [inline projects/desktop/src/components/common/LanguageChooser/LanguageChooser.tsx]: Return function type

File: projects/desktop/src/components/common/LanguageChooser/LanguageChooser.tsx [hunk 20]
```diff
@@ -0,0 +1,58 @@
...
+    const [state, dispatch] = useReducer(reducer, initialState)
+    const {i18n} = useTranslation()
+
+    const toggleDialog = () => {
```
Comment (Sylv11) [inline projects/desktop/src/components/common/LanguageChooser/LanguageChooser.tsx]: Type ?

File: projects/desktop/src/components/common/LanguageChooser/LanguageChooser.tsx [hunk 21]
```diff
@@ -0,0 +1,58 @@
...
+        dispatch({type: actions.TOGGLE_DIALOG})
+    }
+
+    const setLanguage = async (language: string) => {
```
Comment (Sylv11) [inline projects/desktop/src/components/common/LanguageChooser/LanguageChooser.tsx]: Return function type ?

File: projects/desktop/src/components/common/LanguageChooser/LanguageChooserDialog.tsx [hunk 22]
```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import {
+    Dialog
+} from '@material-ui/core'
+import { useTranslation } from 'react-i18next'
+import {LanguageDialogTitle, LanguageList, LanguageListItem} from './styles'
+
+export const LanguageChooserDialog = ({open, selectedValue, onClose}: ISimpleDialogProps) => {
```
Comment (Sylv11) [inline projects/desktop/src/components/common/LanguageChooser/LanguageChooserDialog.tsx]: Return function type

File: projects/desktop/src/components/common/LanguageChooser/LanguageChooserDialog.tsx [hunk 23]
```diff
@@ -0,0 +1,33 @@
...
+    const [t, i18n] = useTranslation()
+
```
Comment (Sylv11) [inline projects/desktop/src/components/common/LanguageChooser/LanguageChooserDialog.tsx:10]: Types ?

File: projects/desktop/src/constants/validators.ts [hunk 24]
```diff
@@ -0,0 +1,14 @@
+export const required = (value: string): string | undefined  => {
```
Comment (Sylv11) [inline projects/desktop/src/constants/validators.ts]: `export const required = (value: string): string | undefined  => 
    value ? undefined : 'fieldError.required'
`

Check if it works like this

File: projects/desktop/src/constants/validators.ts [hunk 25]
```diff
@@ -0,0 +1,14 @@
...
+    return value ? undefined : 'fieldError.required'
+}
+
```
Comment (Sylv11) [inline projects/desktop/src/constants/validators.ts:3]: Same for the other

File: projects/desktop/src/interfaces/dialogs.d.ts [hunk 26]
```diff
@@ -0,0 +1,5 @@
+interface ISimpleDialogProps {
```
Comment (Sylv11) [inline projects/desktop/src/interfaces/dialogs.d.ts:1]: There are coma

File: projects/desktop/src/interfaces/generics.d.ts [hunk 27]
```diff
@@ -0,0 +1,20 @@
+/*
+  File containing generics typescript interface for type checking
+*/
+
+interface IStringTMap<T> {
```
Comment (Sylv11) [inline projects/desktop/src/interfaces/generics.d.ts:5]: What does it mean ?

File: projects/desktop/src/assets/locales/EN.json (hunk 1 above)
Comment (jvnm-dev) [reply projects/desktop/src/assets/locales/EN.json:1]: Trailing comma are not authorized in json files

Comment (Sylv11) [reply projects/desktop/src/assets/locales/EN.json:1]: Okay, don't clash me

File: projects/desktop/src/Router.tsx [hunk 28]
```diff
@@ -1,9 +1,9 @@
...
-export const Router = () => (
-  <BrowserRouter>
-    <Route exact path='/' component={TestScreen} />
-  </BrowserRouter>
-)
+import React from 'react'
+import {BrowserRouter, Route} from 'react-router-dom'
+import {LoginScreen} from './components/LoginScreen/index'
+
+export const Router = () => (
```
Comment (Sylv11) [inline projects/desktop/src/Router.tsx]: Type maybe ?

File: projects/desktop/src/components/LoginScreen/form/Input.tsx [hunk 29]
```diff
@@ -0,0 +1,93 @@
...
+                visible: !state.visible,
+            }
+        default:
+            throw new Error('Action type not found')
+    }
+}
+
+export const Input = (props: IInputProps): JSX.Element => {
+    const [state, dispatch] = useReducer(reducer, initialState)
+    const [t] = useTranslation()
```
Comment (Sylv11) [inline projects/desktop/src/components/LoginScreen/form/Input.tsx]: Type

File: projects/desktop/src/components/LoginScreen/form/Input.tsx [hunk 30]
```diff
@@ -0,0 +1,93 @@
...
+
+    const { visibilityFilter, invalid, value, ...rest } = props
```
Comment (Sylv11) [inline projects/desktop/src/components/LoginScreen/form/Input.tsx]: Interface

File: projects/desktop/src/components/LoginScreen/styles.tsx [hunk 31]
```diff
@@ -0,0 +1,241 @@
...
+    Box as MaterialBox,
+    TextField,
+    Button,
+} from '@material-ui/core'
+import background from '../../assets/img/background.png'
+import { colors } from '../../constants'
+import {Link} from 'react-router-dom'
+import {LanguageChooser} from '../common/LanguageChooser'
+
+export const Background = styled.div`
```
Comment (Sylv11) [inline projects/desktop/src/components/LoginScreen/styles.tsx:13]: Type for all styled ?

File: projects/desktop/src/interfaces/generics.d.ts (hunk 27 above)
Comment (jvnm-dev) [reply projects/desktop/src/interfaces/generics.d.ts:5]: With this you can make an object with string keys and a T type for the value. Example:
`
    const myObject: IStringTMap<Number> = {'one' : 1}
`
`
    const usersDetails: IStringTMap<UserDetail> = {'username': {...}}
`

File: projects/desktop/src/components/common/LanguageChooser/styles.tsx [hunk 32]
```diff
@@ -0,0 +1,17 @@
+import styled from 'styled-components'
+import {DialogTitle, List, ListItem} from '@material-ui/core'
+
+export const LanguageDialogTitle = styled(DialogTitle)`
```
Comment (Sylv11) [inline projects/desktop/src/components/common/LanguageChooser/styles.tsx:4]: Types

File: projects/desktop/src/interfaces/generics.d.ts (hunk 27 above)
Comment (Sylv11) [reply projects/desktop/src/interfaces/generics.d.ts:5]: Okay I see and why `**[**key: string**]**` ?

Comment (Sylv11) [reply projects/desktop/src/interfaces/generics.d.ts:5]: The crochets

Comment (jvnm-dev) [reply projects/desktop/src/interfaces/generics.d.ts:5]: It is used to isolate the type declaration

File: projects/desktop/src/components/LoginScreen/index.tsx (hunk 15 above)
Comment (jvnm-dev) [reply projects/desktop/src/components/LoginScreen/index.tsx:24]: Yes while the backend is not done

**Review by Sylv11 — APPROVED** [review]

**Pull request merged**